### PR TITLE
feat(module:page-header): add page-header component

### PR DESCRIPTION
**Pull request by HsuanXyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2710

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Ref https://github.com/ant-design/ant-design/pull/13637

### CODE REVIEW

File: components/page-header/nz-page-header.component.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import {
+  ChangeDetectionStrategy,
+  Component, ContentChild,
```
Comment (wendzhue) [inline components/page-header/nz-page-header.component.ts]: Suggestion: wrap these imports to make a consistent code style.

File: components/page-header/doc/index.zh-CN.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+| --- | --- | --- | --- |
+| nzTitle | title 文字 | `string｜TemplateRef<void>` | - |
+| nzSubTitle | subTitle 文字 | `string｜TemplateRef<void>` | - |
+| nzBackIcon | 自定义 back icon | `string｜TemplateRef<void>` | - |
+| nzBack | 返回按钮的点击事件 | `EventEmitter<void>` | - |
+
+### Page header 组成部分
+| 元素 | 说明 |
+| ----- | ----------- | ---- | ------------- |
+| `<nz-page-header-title>` | title 部分 |
```
Comment (wendzhue) [inline components/page-header/doc/index.zh-CN.md]: 如果用了 `nzTitle` 似乎 `<nz-page-header-title>` 就不会被显示, 是否要在文档中说明. 其他属性类似.

File: components/page-header/style/index.less [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+  }
+
+  .@{pageheader-prefix-cls}-content-view {
+    padding-top: 12px;
+  }
+
+  .@{pageheader-prefix-cls}-footer {
+    margin: 0 -8px;
+    padding-top: 24px;
```
Comment (wendzhue) [inline components/page-header/style/index.less]: 如果用户把 `nz-page-header-footer` append 到一个内联元素上, 比如 `<span>`, 这个样式会不生效, 是否需要在 directive 的 meta 信息中写死 `display: block`? 其他元素可能也是一样.

File: components/page-header/doc/index.zh-CN.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+## API
+
+```html
+<nz-page-header nzTitle="Page Title"></nz-page-header>
+```
+
+### nz-page-header
+| 参数 | 说明 | 类型 | 默认值 |
+| --- | --- | --- | --- |
+| nzTitle | title 文字 | `string｜TemplateRef<void>` | - |
```
Comment (wendzhue) [inline components/page-header/doc/index.zh-CN.md]: 不仅仅是 `文字`, 因为还支持 TemplateRef.

**Comment by wendzhue** [comment]
@HsuanXyz This is not blocked now since ant design had released this component. Please check: https://ant.design/components/page-header-cn/.

**Comment by TheAifam5** [comment]
I would like to see it in the new release :D

File: components/page-header/doc/index.zh-CN.md (hunk 4 above)
Comment (HsuanXyz) [reply components/page-header/doc/index.zh-CN.md]: 类似于 `ReactNode ` ，`TemplateRef ` 应该不需要做特殊说明

**Pull request merged**